apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1293 @@
+		}
+		cur = cur->next;
+	}
+	dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+
+	/* entity is needed POST or PUT method */
+
+	if (param->method == WGET_MODE_POST || param->method == WGET_MODE_PUT) {
+		/* content length */
+		if (!param->encoding) {
+			dest = ws_test_wget_strcpy(dest, (char *)param->entity, param);
+			if (dest == NULL) {
+				return WGET_MSG_CONSTRUCT_ERR;
+			}
+		} else { /* chunked param->encoding */
+			post_len = strlen(param->entity);
+			if (post_len > param->buflen) {
+				dest = ws_test_wget_chunksize(dest, param->buflen, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strlencpy(dest, (char *)param->entity, param->buflen, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+			} else {
+				temp = dest;
+				dest = ws_test_wget_chunksize(dest, post_len, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, (char *)param->entity, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				g_normal_clen = dest - temp;
+				dest = ws_test_wget_strlencpy(dest, "0", 1, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				if (g_trailer) {
+					dest = ws_test_wget_strcpy(dest, g_expires, param);
+					if (dest == NULL) {
+						return WGET_MSG_CONSTRUCT_ERR;
+					}
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				g_last_clen = dest - temp;
+			}
+		}
+	}
+	FUNC_EX;
+	return dest - param->buffer;
+}
+
+static int ws_test_wget_socket_connect(struct ws_test_wget_s *ws)
+{
+	FUNC_EN;
+	int sockfd, ret;
+	struct timeval tv;
+	struct sockaddr_in server;
+
+	/* Create a socket */
+	sockfd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (sockfd < 0) {
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+
+	/* Set send and receive timeout values */
+	tv.tv_sec = WEBCLIENT_CONF_TIMEOUT_MSEC / 1000;
+	tv.tv_usec = (WEBCLIENT_CONF_TIMEOUT_MSEC % 1000) * 1000;
+	PRNT("recv timeout is set to (%d.%d)sec", tv.tv_sec, tv.tv_usec);
+	if (setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO,
+				   (struct timeval *)&tv, sizeof(struct timeval)) < 0) {
+		PRNT("ERROR: setsockopt failed");
```

Comment:
Is it okay to do more even if failed to set the socket opt?

Reply:
Changes done as suggested